File: user-service/src/main/java/org/mounanga/userservice/dto/PageModel.java
package org.mounanga.userservice.dto;

import lombok.*;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@ToString
public class PageModel<T> {
    private int page;
    private int totalPages;
    private int size;
    private long totalElements;
    private int numberOfElements;
    private int number;
    private boolean hasContent;
    private boolean hasNext;
    private boolean hasPrevious;
    private boolean isFirst;
    private boolean isLast;
    private List<T> content;
}

File: admin-bank-ui/src/app/dto/security/page.model.ts
import {UserResponse} from "./user.response";

export class PageModel {
  page!: number;
  totalPages!: number;
  size!: number;
  totalElements!: number;
  numberOfElements!: number;
  number!: number;
  hasContent!: boolean;
  hasNext!: boolean;
  hasPrevious!: boolean;
  isFirst!: boolean;
  isLast!: boolean;
  content!: UserResponse[];
}
